src/HeckeTypes.jl

```diff
@@ -312,69 +312,56 @@ end
 Type for rows of sparse matrices, to create one use
 `sparse_row`
 """
-mutable struct SRow{T}
+mutable struct SRow{T, S} # S <: AbstractVector{T}
```

Comment:
Any particular reason you are not making this explicit, i.e.:
```suggestion
mutable struct SRow{T, S <: AbstractVector{T}}
```